lib/presentation/widget/upload_file/upload_file_viewmodel.dart

```diff
@@ -86,6 +94,8 @@ class UploadFileViewModel extends BaseViewModel {
           break;
       }
     });
+
+    Future.delayed(Duration(milliseconds: 500), () => _checkContactPermission());
```

Comment:
What is it for ? You want to check after widget building?

Reply:
yes, basically we have to show the request permission dialog after the widget has finished building layout. tried all solutions in here [https://stackoverflow.com/questions/49466556/flutter-run-method-on-widget-build-complete](https://stackoverflow.com/questions/49466556/flutter-run-method-on-widget-build-complete) but it will end up like this.

![Screen Shot 2021-03-25 at 11 45 09](https://user-images.githubusercontent.com/16035728/112569483-ce824280-8e16-11eb-90f6-2e675141740c.png)

Reply:
cool.

Reply:
It depends to much on users device but if it's the only way to do it let's go